pkg/network/http/shared_libraries.go

```diff
@@ -5,25 +5,172 @@ package http
 import (
 	"fmt"
 	"os"
-	"path/filepath"
+	fp "path/filepath"
 	"regexp"
 	"strings"
+	"time"
+	"unsafe"
 
+	ddebpf "github.com/DataDog/datadog-agent/pkg/ebpf"
+	"github.com/DataDog/datadog-agent/pkg/util/log"
+	"github.com/DataDog/gopsutil/process/filepath"
 	"github.com/DataDog/gopsutil/process/so"
 )
 
-const (
-	libssl    = "libssl.so"
-	libcrypto = "libcrypto.so"
-)
+/*
+#include "../ebpf/c/http-types.h"
+*/
+import "C"
 
-var openSSLLibs = regexp.MustCompile(
-	fmt.Sprintf("(%s|%s)", regexp.QuoteMeta(libssl), regexp.QuoteMeta(libcrypto)),
-)
+const pathMaxSize = int(C.LIB_PATH_MAX_SIZE)
+
+type libPath = C.lib_path_t
```

Comment:
Not required for this PR, but would be nice to use `cgo` generated types.

Reply:
I'll update it